src/Features/Core/Portable/ReplaceMethodWithProperty/AbstractReplaceMethodWithPropertyService.cs

```diff
@@ -47,18 +47,35 @@ private static bool OverridesMetadataSymbol(IMethodSymbol method)
         {
             var getMethodDeclaration = getAndSetMethods.GetMethodDeclaration;
             var setMethodDeclaration = getAndSetMethods.SetMethodDeclaration;
-            if (setMethodDeclaration == null)
+            var finalLeadingTrivia = getAndSetMethods.GetMethodDeclaration.GetLeadingTrivia().ToList();
+
+            //If there is a comment on the same line as the method it is contained in trailing trivia for the parameter list
+            //If it's there we need to add it to the final comments
+            //this is to fix issue 42699, https://github.com/dotnet/roslyn/issues/42699
+            if (getMethodDeclaration.ChildNodes().Any(n => n.RawKind == syntaxFacts.SyntaxKinds.ParameterList && n.GetTrailingTrivia().Any(t => !syntaxFacts.IsWhitespaceOrEndOfLineTrivia(t))))
             {
-                return property.WithLeadingTrivia(getMethodDeclaration.GetLeadingTrivia());
+                //we have a meaningful comment on the parameter list so add it to the trivia list
+                finalLeadingTrivia.AddRange(getMethodDeclaration.ChildNodes().Where(n => n.RawKind == syntaxFacts.SyntaxKinds.ParameterList).First().GetTrailingTrivia());
             }
 
-            var finalLeadingTrivia = getAndSetMethods.GetMethodDeclaration.GetLeadingTrivia().ToList();
+            if (setMethodDeclaration == null)
+            {
+                return property.WithLeadingTrivia(finalLeadingTrivia);
+            }
 
             finalLeadingTrivia.AddRange(
                 setMethodDeclaration.GetLeadingTrivia()
                                     .SkipWhile(t => syntaxFacts.IsEndOfLineTrivia(t))
                                     .Where(t => !t.IsDirective));
 
+            //If there is a comment on the same line as the method it is contained in trailing trivia for the parameter list
+            //If it's there we need to add it to the final comments
+            if (setMethodDeclaration.ChildNodes().Any(n => n.RawKind == syntaxFacts.SyntaxKinds.ParameterList && n.GetTrailingTrivia().Any(t => !syntaxFacts.IsWhitespaceOrEndOfLineTrivia(t))))
```

Comment:
Untested, but I think `syntaxFacts.GetParameterList(setMethodDeclaration)` should work here, and it would be more readable to introduce a local for the result of that, and go from there.

Reply:
Yes, GetParameterList works. I have changed the code to utilize that function.